src/udisks2blockdevices.cpp

```diff
@@ -51,62 +51,72 @@ BlockDevices *BlockDevices::instance()
 
 BlockDevices::~BlockDevices()
 {
-    qDeleteAll(m_blockDevices);
+    qDeleteAll(m_activeBlockDevices);
     qDeleteAll(m_partitionWaits);
 
-    m_blockDevices.clear();
+    m_activeBlockDevices.clear();
     m_partitionWaits.clear();
     sharedInstance = nullptr;
 }
 
 bool BlockDevices::contains(const QString &dbusObjectPath) const
 {
-    return m_blockDevices.contains(dbusObjectPath);
+    return m_activeBlockDevices.contains(dbusObjectPath);
 }
 
 void BlockDevices::remove(const QString &dbusObjectPath)
 {
     if (contains(dbusObjectPath)) {
-        Block *block = m_blockDevices.take(dbusObjectPath);
+        Block *block = m_activeBlockDevices.take(dbusObjectPath);
+        m_blockDevices.remove(dbusObjectPath);
         clearPartitionWait(dbusObjectPath, false);
         delete block;
     }
 }
 
 Block *BlockDevices::device(const QString &dbusObjectPath) const
 {
+    Block *block = m_activeBlockDevices.value(dbusObjectPath, nullptr);
+    if (block) {
+        return block;
+    }
     return m_blockDevices.value(dbusObjectPath, nullptr);
 }
 
 Block *BlockDevices::replace(const QString &dbusObjectPath, Block *block)
```

Comment:
It feels to me that `replace` no longer describes this function.

Reply:
Yes it really doesn't. Will rename to deactivate.